src/yz_events.erl

```diff
@@ -59,7 +70,26 @@
 %%%===================================================================
 
 start_link() ->
-    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+    gen_event:start_link({local, ?MODULE}).
+
+add_handler(Handler, Args) ->
+    gen_event:add_handler(?MODULE, Handler, Args).
+
+add_sup_handler(Handler, Args) ->
+    gen_event:add_sup_handler(?MODULE, Handler, Args).
+
+add_guarded_handler(Handler, Args) ->
+    yz_eventhandler_sup:start_guarded_handler(?MODULE, Handler, Args).
+
+add_callback(Fn) when is_function(Fn) ->
+    gen_event:add_handler(?MODULE, {?MODULE, make_ref()}, [Fn]).
+
+add_sup_callback(Fn) when is_function(Fn) ->
+    gen_event:add_sup_handler(?MODULE, {?MODULE, make_ref()}, [Fn]).
+
+add_guarded_callback(Fn) when is_function(Fn) ->
+    yz_eventhandler_sup:start_guarded_handler(?MODULE, {?MODULE, make_ref()},
+                                          [Fn]).
```

Comment:
A lot of these add_ calls are not actually used in the code, except current, add_guarded_handler.  We should either remove these, if they are not needed, or document why they are there.